skyportal/models/analysis.py

```diff
@@ -152,3 +187,196 @@ def authinfo(self):
+            raise ValueError(f'Illegal characters in string "{string}". ')
+
+    @hybrid_property
+    def data(self):
+        """Lazy load the data dictionary"""
+        if self._data is None:
+            self.load_data()
+        return self._data
+
+    _unique_id = sa.Column(
+        sa.String,
+        nullable=False,
+        unique=True,
+        default=lambda: str(uuid.uuid4()),
+        doc='Unique identifier for this analysis result.',
+    )
+
+    hash = sa.Column(
+        sa.String,
+        nullable=True,
+        unique=True,
+        doc='MD5sum hash of the data to be saved to file. Prevents duplications.',
+    )
+
+    show_parameters = sa.Column(
+        sa.Boolean,
+        default=False,
+        nullable=False,
+        doc="Whether to render the parameters of this analysis",
+    )
+
+    show_plots = sa.Column(
+        sa.Boolean,
+        default=False,
+        nullable=False,
+        doc="Whether to render the plots of this analysis",
+    )
+
+    show_corner = sa.Column(
+        sa.Boolean,
+        default=False,
+        nullable=False,
+        doc="Whether to render the corner plots of this analysis",
+    )
+
+    analysis_parameters = sa.Column(
+        JSONType,
+        nullable=True,
+        doc=('Optional parameters that are passed to the analysis service'),
+    )
+
+    @classmethod
+    def backref_name(cls):
+        if cls.__name__ == 'ObjAnalysis':
+            return "obj_analyses"
+
+    @declared_attr
+    def author_id(cls):
+        return sa.Column(
+            sa.ForeignKey('users.id', ondelete='CASCADE'),
+            nullable=False,
+            index=True,
+            doc="ID of the Annotation author's User instance.",
+        )
+
+    @declared_attr
+    def author(cls):
+        return relationship(
+            "User",
+            doc="Annotation's author.",
+        )
+
+    @declared_attr
+    def analysis_service_id(cls):
+        return sa.Column(
+            sa.ForeignKey('analysis_services.id', ondelete='CASCADE'),
+            nullable=False,
+            index=True,
+            doc="ID of the associated analysis service.",
+        )
+
+    @declared_attr
+    def analysis_service(cls):
+        return relationship(
+            "AnalysisService",
+            back_populates=cls.backref_name(),
+            doc="Analysis Service associated with this analysis.",
+        )
+
+    @declared_attr
+    def groups(cls):
+        return relationship(
+            "Group",
+            secondary="group_" + cls.backref_name(),
+            cascade="save-update, merge, refresh-expire, expunge",
+            passive_deletes=True,
+            doc="Groups that can see the analysis.",
+        )
+
+    def construct_creator_info_dict(self):
+        return {
+            field: getattr(self.author, field)
+            for field in ('username', 'first_name', 'last_name', 'gravatar_url')
+        }
+
+
+class ObjAnalysis(Base, AnalysisMixin, WebhookMixin):
```

Comment:
I guess the plan is to have Analysis on spectrum, and so on?

Reply:
Yes